bindings/c/test/apitester/TesterAtomicOpsCorrectnessWorkload.cpp

```diff
@@ -0,0 +1,368 @@
+	void testAtomicVersionstampedKeyOp(TTaskFct cont) {
+		Key keyPrefix(randomKeyName() + getKeySuffix());
+		Key key = keyPrefix + fdb::ByteString(10, '\0') + toByteString((uint32_t)keyPrefix.size());
+		Value val = randomValue();
+
+		auto versionstamp = std::make_shared<Key>();
+		execTransaction(
+		    // 1. Perform SetVersionstampedKey operation.
+		    [key, val, versionstamp](auto ctx) {
+			    ctx->tx().atomicOp(key, val, FDBMutationType::FDB_MUTATION_TYPE_SET_VERSIONSTAMPED_KEY);
+			    // TODO: Figure out how to correctly retrieve version timestamps.
+			    // See testAtomicVersionstampedValueOp() for detailed discussion.
+			    ctx->commit();
+		    },
+		    [this, keyPrefix, val, versionstamp, cont]() {
+			    auto resultKey = std::make_shared<Key>();
+			    auto resultVal = std::make_shared<Value>();
+			    execTransaction(
+			        // 2. Fetch the resulting versionstamped key and value.
+			        [keyPrefix, resultKey, resultVal](auto ctx) {
+				        auto f = ctx->tx().getKey(fdb::key_select::firstGreaterOrEqual(keyPrefix), false);
+				        ctx->continueAfter(
+				            f,
+				            [ctx, f, resultKey, resultVal]() {
+					            *resultKey = f.get();
+					            auto fv = ctx->tx().get(*resultKey, false);
+					            ctx->continueAfter(fv, [ctx, fv, resultVal]() {
+						            auto outputVal = fv.get();
+						            ASSERT(outputVal.has_value());
+						            *resultVal = outputVal.value();
+
+						            ctx->done();
+					            });
+				            },
+				            true);
+			        },
+			        [this, keyPrefix, val, resultKey, resultVal, versionstamp, cont]() {
+				        // 3. Assert expectation.
+				        // ASSERT(*resultKey == keyPrefix + *versionstamp);
+				        // TODO: Enable the above assert instead of the ones below once versionstamp
+				        // is correctly retrieved.
+				        ASSERT(resultKey->substr(0, keyPrefix.size()) == keyPrefix);
+				        ASSERT(toUint64_t(resultKey->substr(keyPrefix.size(), 8)) > 0);
+				        ASSERT(*resultVal == val);
+				        schedule(cont);
+			        });
+		    });
+	}
+
+	void testAtomicVersionstampedValueOp(TTaskFct cont) {
+		Key key(randomKeyName() + getKeySuffix());
+		Value valPrefix = randomValue();
+		Value val = valPrefix + fdb::ByteString(10, '\0') + toByteString((uint32_t)valPrefix.size());
+		auto versionstamp = std::make_shared<Key>();
+		execTransaction(
+		    // 1. Perform SetVersionstampedValue operation.
+		    [key, val, versionstamp](auto ctx) {
+			    ctx->tx().atomicOp(key, val, FDBMutationType::FDB_MUTATION_TYPE_SET_VERSIONSTAMPED_VALUE);
+			    // TODO: Figure out how to correctly retrieve version timestamps. This may require
+			    // changes to TesterTransactionExectutor.
+			    // The following works most of the time, but fails when the transaction returns
+			    // commit_unknown_result, and the TransactionExecutor automatically retries it.
```

Comment:
Let's introduce an option something like retryOnCommitUnknownResult to the TransactionExecutor

Reply:
Yes, but that applies to the comment above. This particular problem should be fixed differently. TransactionExecutor shouldn't be failing retriable transactions.